### PR TITLE
1 feature 2 project manager task initialization

### PR DESCRIPTION
**Pull request by jianwang0722**
created classes for Use Case project manager task initialization, added entity commonTaskFactory, interactor and interface adapter layer still incomplete

Edit: basic complete use case classes implemented.

### CODE REVIEW

**Review by zeyuzhang516 — DISMISSED** [review]

**Review by GitHaoBao — DISMISSED** [review]

File: src/main/java/entity/CommonTaskFactory.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package entity;
+
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
+public class CommonTaskFactory {
```
Comment (tlan4) [inline src/main/java/entity/CommonTaskFactory.java:8]: Should have an interface

File: src/main/java/presenter/LoginResponsePresenter.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package Presenter;
+package presenter;
 
 import Entity.User;
```
Comment (tlan4) [inline src/main/java/presenter/LoginResponsePresenter.java:3]: entity

Comment (jianwang0722) [reply src/main/java/presenter/LoginResponsePresenter.java:3]: Oh this is actually Josh's part, I accidentally changed it, oops!

File: src/main/java/entity/CommonTaskFactory.java (hunk 1 above)
Comment (jianwang0722) [reply src/main/java/entity/CommonTaskFactory.java:8]: Thx, will add

Comment (tlan4) [reply src/main/java/entity/CommonTaskFactory.java]: I added in #34

**Review by kaizen-w — APPROVED** [review]
Looks good and follows clean architecture.

File: src/main/java/entity/CommonTaskFactoryInterface.java [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package entity;
+
+public interface CommonTaskFactoryInterface {
```
Comment (tlan4) [inline src/main/java/entity/CommonTaskFactoryInterface.java:3]: I don't think this file is needed

File: src/main/java/project_manager_task_init_use_case/PMTaskInitInteractor.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.Set;
+
+public class PMTaskInitInteractor implements PMTaskInitInputBoundary{
+
+    final PMTaskInitOutputBoundary presenter;
+
+    final CommonTaskFactory commonTaskFactory;
```
Comment (tlan4) [inline src/main/java/project_manager_task_init_use_case/PMTaskInitInteractor.java:16]: Use the general TaskFactory as interface

File: src/main/java/entity/CommonTaskFactoryInterface.java (hunk 3 above)
Comment (jianwang0722) [reply src/main/java/entity/CommonTaskFactoryInterface.java:3]: You are correct, will remove it when I push my next changes

Comment (tlan4) [reply src/main/java/entity/CommonTaskFactoryInterface.java:3]: Should use the general taskfactory as interface, and the commontaskfactory as the concrete object in the constructor when you are using the interactor

File: src/main/java/project_manager_task_init_use_case/PMTaskInitInteractor.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+    final PMTaskInitGateway taskInitGateway;
+
+    public PMTaskInitInteractor(PMTaskInitOutputBoundary presenter, CommonTaskFactory commonTaskFactory, PMTaskInitGateway gateway) {
```
Comment (tlan4) [inline src/main/java/project_manager_task_init_use_case/PMTaskInitInteractor.java:20]: Don't use concrete CommonTaskFactory, use general TaskFactory

**Review by tlan4 — CHANGES_REQUESTED** [review]

File: src/main/java/project_manager_task_init_use_case/PMTaskInitInteractor.java (hunk 5 above)
Comment (jianwang0722) [reply src/main/java/project_manager_task_init_use_case/PMTaskInitInteractor.java:20]: Yes, I have already changed it

File: src/main/java/project_manager_task_init_use_case/PMTaskInitInteractor.java (hunk 4 above)
Comment (jianwang0722) [reply src/main/java/project_manager_task_init_use_case/PMTaskInitInteractor.java]: Done

**Review by tlan4 — APPROVED** [review]

**Review by staceylry — APPROVED** [review]

**Review by KunLW — APPROVED** [review]

**Review by GitHaoBao — APPROVED** [review]

**Review by lungerjo — APPROVED** [review]

**Pull request merged**